### DISCUSSION TITLE
Contributing and 3.0 ship date

### DISCUSSION THREAD

**Discussion by adv12** [tomchavakis/nuget-license#185] [General]
Hi.  I see there's a 3.0 alpha branch with lots of new work on it.  I would like to make one or more contributions to this project, but it looks as if maybe both 2.X and 3.0 have stalled a bit.  I'm wondering where, if anywhere, I should submit a pull request to get it into future versions of the shipping product.  I don't know git very well, so it's unclear to me how conflicting changes in the two branches will be resolved when the merge happens.

**Comment by adv12** [comment]
I should clarify that my goals are 1) expediency and 2) longevity.  I want to make a change that will ship soon and also be retained in future versions.

**Reply by sensslen** [reply]
Hi @adv12,

First of all: sorry for the long wait - your request has slipped through (for me at least).
You are right about stalled development in the sense as there was no need for further additions to the 3.x version. If you ask me (I've been the driving force behind version 3, as I needed a reliable way to resolve exact versions and not ranges), 3.x should be ready to ship. However that's a decision @tomchavakis will have to make. Contributions to both versions are certainly very welcome.
